packages/marketplace/src/components/pages/app-detail/client/client-web-component-config.tsx

```diff
@@ -1,36 +1,48 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
 import { Button } from '@reapit/elements'
 import { useDispatch, useSelector } from 'react-redux'
-import { selectIsWebComponentOpen } from '@/selector/client'
-import { clientCloseWebComponentConfig, clientOpenWebComponentConfig } from '@/actions/client'
+import { selectWebComponentData } from '@/selector/client'
+import { clientFetchWebComponentConfig } from '@/actions/client'
 import { Dispatch } from 'redux'
 import WebComponentModal from '@/components/ui/web-component-config-modal'
 import { AppDetailSection } from '../common/ui-helpers'
+import { selectClientId } from '@/selector/auth'
+import { useParams } from 'react-router-dom'
 
-export const toggleWebComponentModal = (dispatch: Dispatch) => () => {
-  dispatch(clientOpenWebComponentConfig())
+export const toggleWebComponentModal = (setIsOpenConfigModal, isOpen) => () => {
+  setIsOpenConfigModal(isOpen)
 }
 
-export const closeWebComponentModal = (dispatch: Dispatch) => () => {
-  dispatch(clientCloseWebComponentConfig())
+export const handleFetchWebComponentConfig = (
+  dispatch: Dispatch,
+  customerId?: string,
+  applicationId?: string,
+) => () => {
+  customerId && applicationId && dispatch(clientFetchWebComponentConfig({ customerId, applicationId }))
 }
 
 export const WebComponentConfig: React.FC = () => {
   const dispatch = useDispatch()
-  const isWebComponentOpen = useSelector(selectIsWebComponentOpen)
-  const handleToggleWebComponentModal = toggleWebComponentModal(dispatch)
-  const handleCloseWebComponentModal = closeWebComponentModal(dispatch)
+  const [isOpenConfigModal, setIsOpenConfigModal] = useState(false)
+
+  const clientId = useSelector(selectClientId) || ''
+  const webComponentData = useSelector(selectWebComponentData)
+  const { appid: applicationId } = useParams()
```

Comment:
`appid` could we change to `appId` ?

Reply:
Yes, but I think we should change it in separate ticket because all routes use "appid" params now.